pkg/api/server.go

```diff
@@ -79,6 +79,11 @@ func (a *ApiServer) Start() error {
 	docs.SwaggerInfo.Description = "Daytona Server API"
 	docs.SwaggerInfo.Title = "Daytona Server API"
 
+	_, err := net.Dial("tcp", fmt.Sprintf(":%d", a.apiPort))
+	if err == nil {
+		log.Errorf("Cannot start API server, port %d is already in use", a.apiPort)
```

Comment:
These errors need to be returned from the Start function. Your way, the execution would just continue and it would show an error twice.